src/EventHandler/Channel/ChannelParticipant.php

```diff
@@ -63,13 +62,13 @@ public function __construct(MTProto $API, array $rawChannelParticipant)
         $this->inviteLink = isset($rawChannelParticipant['invite'])
             ? ChatInvite::fromRawChatInvite($rawChannelParticipant['invite'])
             : null;
-        // If null, user lefted channel
+        // If null, user wasn't a participant.
         $this->prevParticipant = isset($rawChannelParticipant['prev_participant'])
             ? Participant::fromRawParticipant($rawChannelParticipant['prev_participant'])
             : new Left(['peer' => $this->userId]);
-        // if null, user joind
+        // If null, user has left.
         $this->newParticipant = isset($rawChannelParticipant['new_participant'])
             ? Participant::fromRawParticipant($rawChannelParticipant['new_participant'])
-            : new Member(['user_id' => $this->userId, 'date' => $this->date]);
+            : new Left(['peer' => $this->userId]);
```

Comment:
I do not understand why are you doing this change here.

Reply:
Not sure which part you have trouble with, but lemme start from the base.

The `prev_participant`/`new_participant` fields in an update show the status (member, left, etc.) of a user in channel. If either of them is **not set**, means the user is not part of the channel. Either **before joining** or **after leaving**. Right?!

---

This section checks to see if `prev_participant` field is present in the update:

```php
$this->prevParticipant = isset($rawChannelParticipant['prev_participant'])
    ? Participant::fromRawParticipant($rawChannelParticipant['prev_participant'])
    : new Left(['peer' => $this->userId]);
```

- If set, we good. User was a member/admin in the channel.
- If **not set**, user has just joined.

> Nooo problem, till here!

---

The same for `new_participant`, checking for its presence:

```php
$this->newParticipant = isset($rawChannelParticipant['new_participant'])
    ? Participant::fromRawParticipant($rawChannelParticipant['new_participant'])
    : new Member(['user_id' => $this->userId, 'date' => $this->date]);
```

- If set, again, we good. User is a new member/admin.
- If **not set**, user has left the channel.

One would expect the "**new**" status to be `Left` when a member leaves the channel. Yet, the current behavior gives us `Member`. Meaning MadelineProto FAILS to tell you that a member has left.

Reply:
Thank you, could you please add this text to the documentation of the two properties?